xonsh/history.py

```diff
@@ -432,26 +621,30 @@ def _gc(ns, hist):
 
 
 _MAIN_ACTIONS = {
-    'show': _show,
+    'session': History.show,
```

Comment:
@scopatz I will add it back in on popular request, but I encourage you to consider its utility (backwards compatibility) versus its downsides. I would like others to weigh in as well.

As I stated it is redundant as `history` will perform the same functionality. This should be evidence enough for it removal. In addition, with the expanded functionality `show` is ambiguous. Show what? `session` and `all` clarify that ambiguity. For these reasons I am strongly against restoring it, in the name of logic and cleanliness.

In what way are you using `show` that would justify leaving in ambiguous redundancy? If you need the default `session` behavior (which mostly mimics the old `show`) changed to be more acceptable I am open to discussion.